docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -291,6 +305,42 @@ type LocalObjectTemplate struct {
   // offered by a provider.
   Ref *corev1.ObjectReference `json:"ref"`
 }
+
+// MachineHealthCheckClass defines a MachineHealthCheck for a group of Machines.
+type MachineHealthCheckClass struct {
```

Comment:
The fields below seem to be mostly duplicates of the fields in `MachineHealthCheckSpec`. Are we concerned this might cause some trouble long-term for maintenance? How do we ensure the two don't diverge? What if a new feature is added to one place but not the other? Have we considered sharing a common struct so they can share the fields that are in both places or making one a subset of the other?

Please let me know if this comment feels out of scope. I realize this is a pattern that exists in other places in ClusterClasses, which is why this may be a bigger discussion.

Reply:
+1 to start working on some guidelines for common structs, given that this was discussed also in last week office hours